### DISCUSSION TITLE
Desklet reload failes with "Type name Gjs_{myclassname}" is already registered.

### DISCUSSION THREAD

**Discussion by jtoberling** [linuxmint/discussions#181] [Development] [answered]
I'm developing a Cinnamon desklet app - that is extending St.Button with some additional data and from St.Button inherits GObject registration.
During development I used to reload desklet. Since extending the Gjs class the reload does not work just only complete restart of Cinnamon Environment due to during reloading the system is complaining that the Gjs_class is already registered from the previous start.

IMHO there should be a mechanism exists to prevent such error as Reloading could not be happen and this is "not the fault" of the instance issued by the 2nd reload however such feature should be guaranteed by the framework. 
And/or at least a mechanism should exists where I can "unregister" self Gjs_* extension classes during the destroy of application (desklet in my case).

Reference: [yarr/linkbutton.js](https://github.com/linuxmint/cinnamon-spices-desklets/blob/master/yarr%40jtoberling/files/yarr%40jtoberling/linkbutton.js)

**Comment by mtwebster** [comment, marked as answer]
You can't unload/reload GObject types (this isn't to do with gjs, but a behavior of GObject).  You'll have to use delegation.

like...
```
const YarrLinkButton = class {
    constructor() {
        this.uri = null;
        this.actor = new St.Button();
    }
    ...
});
```

**Comment by jtoberling** [comment]
Thanks for pointing on this. By using delegation and adding YarrLinkButton.getActor I was able to change the code to work ok on reload. 

Moreover I could realize I can add extra info to click event and use standard St.Button class without extension. :)

`...`
`             this._signals.connect( feedButton, 'clicked', (...args) => this.onClickedButton(...args, 'my_additional_info') );`
`...`
